openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx

```diff
@@ -249,6 +291,203 @@ const GlossaryTermTab = ({
     return data;
   }, [glossaryTerms, permissions]);
 
+  const listOfVisibleColumns = useMemo(() => {
+    return ['name', 'description', 'owners', 'status', 'new-term'];
+  }, [permissions]);
+
+  const defaultCheckedList = useMemo(() => {
+    return columns
+      .filter((column) => listOfVisibleColumns.includes(column.key as string))
+      .map((column) => column.key)
+      .filter((key): key is string => key !== undefined);
+  }, [columns, listOfVisibleColumns]);
+
+  const [checkedList, setCheckedList] = useState<string[]>(defaultCheckedList);
+  const [tempCheckedList, setTempCheckedList] = useState<string[]>([
+    ...checkedList,
+  ]);
+
+  const [isDropdownVisible, setIsDropdownVisible] = useState<boolean>(false);
+
+  const [options, setOptions] = useState<{ value: string; label: string }[]>(
+    columns
+      .map(({ key, title }) => ({
+        label: title as string,
+        value: key as string,
+      }))
+      .filter((col) => col.value !== 'name')
```

Comment:
Same here let's use `.reduce`

Reply:
Done. Refactored the code to use .reduce()